src/main/java/uk/gov/companieshouse/accounts/association/utils/RequestContextUtil.java

```diff
@@ -0,0 +1,22 @@
+package uk.gov.companieshouse.accounts.association.utils;
+
+import java.util.Objects;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+public class RequestContextUtil {
+
+    private static final String X_REQUEST_ID = "X-Request-Id";
+
+    public static String getXRequestId() {
+        try {
+            final var requestAttributes = RequestContextHolder.getRequestAttributes();
+            final var servletRequestAttributes = ( (ServletRequestAttributes) requestAttributes );
+            final var httpServletRequest = Objects.requireNonNull( servletRequestAttributes ).getRequest();
+            return httpServletRequest.getHeader( X_REQUEST_ID );
+        } catch ( Exception e ){
+            return null;
```

Comment:
this will throw sonar issue cannot return null. you need to handle the exception gracefully.

Reply:
is this thread safe?
for example will it stay the same for the same request and change for different requests?
We need to track the same request id all through the journey of a request.
will the request id coming from front end? or is it generated as soon as it hits the endpoint?

Reply:
Good point - I'll change it so that it doesn't return null. 

FYI - in practice the request will always contain an X-Request-Id, because the request always comes via Eric - so the null shouldn't really be ever returned (also for accounts-association-api, X-Request-Id is mandatory, so is guaranteed to be there). 

This is thread safe because the RequestContextHolder uses ThreadLocal to store the request data. Same approach we used for the UserContext.